### DISCUSSION TITLE
Mechanics?

### DISCUSSION THREAD

**Discussion by yusuf-jkhan1** [Azure/azure-storage-fuse#1098] [General]
Not to overly overly simplify, but I was looking at the dependencies of the repo and saw azcopy and fsnotify. Are those the key components? If so that's awesome because I literally have been hacking around with a similar pattern at work using Watchdog (which wraps different fsmonitor type implementations) and having file modified events trigger a call back to azcopy sync. Thinking of putting it it to a queue instead and using purely copy to minimize scanning, and only do a sync periodically.

**Comment by bhansalivikas** [comment]
Blobfuse2 integrates with fsnotify for the health monitor purpose where we emit notifications on when a file was created by blobfuse2 in local cache and when it was deleted.
Integration with AzCopy is due to the fact that some of the SDK pieces to connect to Azure Storage reside in AzCopy repo.
That being said, what you are trying to achieve is not a work-flow that blobfuse solves. For event based or time based sync you need to use AzCopy and trigger it manually through other scripts.
